### PR TITLE
Replacing Gridicon assets with Gridicons-iOS

### PR DESCRIPTION
**Pull request by frosty**
This PR removes all image assets with 'gridicon' in the name and uses Gridicons-iOS to generate them at runtime.

To test:

* Check that the replaced images appear okay (most are in the blog detail menu). They should all be tinted, so should have the correct appearance.
* Definitely check an instance of `WPWebViewController`, as those buttons are created in Interface Builder but the images are set at runtime.
* Run the `review-deleted-images-for-commit` script to check the images aren't used anywhere.

@mattmiklic Would you mind checking that these look okay to you from a design point of view? The images are all displayed at default size, except for `gridicons-notice` was in the project at 20x20 (so I've stuck with that size for now). Not sure if you'll want to update this to 24?

Needs Review: @kurzee

Thank you!

### CODE REVIEW

**Comment by mattmiklic** [comment]
> The images are all displayed at default size, except for gridicons-notice was in the project at 20x20 (so I've stuck with that size for now)

I had to ping @aerych to remember why this one's used at that size -- this image is right-aligned with the system disclosure indicator `>` image so the default whitespace within gridicons was removed. I actually think we should be using a different name for this image, since it is not a standard gridicon. I would rename the existing image to something like `sharing-notice`, and keep it with the other static image assets, then create a new `gridicons-notice` at the standard 24x24 using the [original unmodified image](https://github.com/Automattic/gridicons/blob/master/pdf/gridicons-notice.pdf).

**Comment by aerych** [comment]
Those darn special cases

**Comment by kurzee** [comment]
@frosty this is great stuff. What do you think about the pod instead being a more general "WPiOS Icons" pod/project? I might find myself wanting to put other assets in an external pod/project as well to follow the same pattern, maybe?

**Comment by kurzee** [comment]
Reviewed all the deletions and updates, looks good. Will review again with the modified icon mentioned by @mattmiklic

**Comment by frosty** [comment]
Thanks both! I've put the sharing notice icon back in as an asset, named `sharing-notice`. If you want to test this, you can comment out lines 201 and 204 in `SharingViewController.m` and then view the Sharing screen for a site that has sharing connections set up.

File: WordPress/Classes/ViewRelated/Blog/WPStyleGuide+Sharing.swift [hunk 1]
```diff
@@ -16,8 +17,7 @@ extension WPStyleGuide
         let horizontalPadding = 8.0;
         let imageView = UIImageView(frame:CGRect(x: 0, y: 0, width: imageSize + horizontalPadding, height: imageSize))
 
-        let noticeImage = UIImage(named: "gridicons-notice")
-        imageView.image = noticeImage?.imageWithRenderingMode(.AlwaysTemplate)
+        imageView.image = UIImage(named: "sharing-notice")
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/WPStyleGuide+Sharing.swift]: @frosty I think we need to keep the call to `imageWithRenderingMode` here.

Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/WPStyleGuide+Sharing.swift]: @aerych I set the image as 'template' in the asset catalog: https://github.com/wordpress-mobile/WordPress-iOS/pull/5096/commits/591fc330d18aa586cb134e75f62303aa12e9dabd#diff-7eef83a33166c97c4ab94e829cd5f91bR13

Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/WPStyleGuide+Sharing.swift]: Oh nice! I was unaware of this wizardry :)

**Comment by kurzee** [comment]
Verified the asset change, looks good.

screen shot 2016-04-07 at 11 14 34 am

:shipit: @frosty!

**Pull request merged**